Gill_Kyle/details.py

```diff
@@ -0,0 +1,24 @@
+from django.conf import settings
+from django_mako_plus import view_function, jscontext
+from django.shortcuts import render
+from catalog import models as cmod
+import math
+
+
+@view_function
+def process_request(request, product_id):
+    selected_product = cmod.Product.objects.filter(id=product_id).first()
```

Comment:
I would suggest having initialization of selected_product in a try except block so the site is not destroyed if an invalid product id is in the /detail url

Reply:
This actually would have saved a lot of problems, don't know why I don't think of that before.